Src/HALALMock/Services/Communication/Ethernet/UDP/DatagramSocket.cpp

```diff
@@ -43,9 +24,44 @@ DatagramSocket::~DatagramSocket(){
 	if(not is_disconnected)
 		close();
 }
-
+DatagramSocket::create_udp_socket(){
+	udp_socket = socket(AF_INET,SOCK_DGRAM,0);
+	if(udp_socket < 0){
+		std::cout<<"Socket creation failed\n";
+	}
+	struct sockaddr_in servaddr; 
+	servaddr.sin_family = AF_INET; 
+	servaddr.sin_port = htons(local_port); 
+	servaddr.sin_addr.s_addr = local_ip.address; 
+	if(bind(udp_socket, (struct sockaddr*)&servaddr, sizeof(servaddr)) < 0){
+		std::cout<<"Bind error\n";
+		close(udp_socket);
+		is_disconnected = true;
+		return;
+	}
+	is_disconnected = false;
+	//receiving callback
+	receiving_udp_thread = std::jthread [&](){
+		is_receiving = true;
+		while(true){
+			uint8_t received_data[1024];
+			struct sockaddr_in src_addr;
+			socklen_t addr_len = sizeof(src_addr);
+			ssize_t size = recvfrom(udp_socket,(uint8_t*)received_data,MAX_SIZE_PACKET,0,(struct sockaddr *)&src_addr, &addr_len);
+			if(size < 0){
+				cout<<"Error in function recvfrom\n";
+				is_receiving = false;
+				return;
+			}
```

Comment:
don't know if size can be negative, what is `ssize_t`?

Reply:
![image](https://github.com/user-attachments/assets/ce167918-b3b5-4105-a3a4-135cc76281fd)